src/Kiota.Builder/Refiners/CommonLanguageRefiner.cs

```diff
@@ -122,29 +120,32 @@ private static bool DoesAnyParentHaveAPropertyWithDefaultValue(CodeClass current
                 AccessedProperty = currentProperty,
             }).First();
             currentProperty.Getter.Name = $"{getterPrefix}{accessorName}"; // so we don't get an exception for duplicate names when no prefix
-            var setter = parentClass.AddMethod(new CodeMethod {
-                Name = $"set-{accessorName}",
-                Access = AccessModifier.Public,
-                IsAsync = false,
-                Kind = CodeMethodKind.Setter,
-                Description = $"Sets the {propertyOriginalName} property value. {currentProperty.Description}",
-                AccessedProperty = currentProperty,
-                ReturnType = new CodeType {
-                    Name = "void",
-                    IsNullable = false,
-                    IsExternal = true,
-                },
-            }).First();
-            setter.Name = $"{setterPrefix}{accessorName}"; // so we don't get an exception for duplicate names when no prefix
-            currentProperty.Setter = setter;
+            if (!currentProperty.IsOfKind(CodePropertyKind.BackingStore))
```

Comment:
@baywet your thoughts on NOT adding setters for backing store properties?

Reply:
I think the idea behind the setter was that if you already have a context with some data (e.g. fluid app "waking up") you'll need to re-hydrate your models from the context, and potentially set the backing store on a new model.

Reply:
I'd think it would be better that a new model only initialises its backing store once via the constructor.
Unless we want to support scenarios where different instances of a model use different types of backing stores..?

Reply:
not all languages support constructor overloads (TS, Go...). Which means we'd have to make the backing store an optional parameter (when supported, Go doesn't), which would make simple scenarios like newing up a model a bit more complex and deserialization of complex properties as well.
Having the backing store being swapped during a model's lifetime is not the end of the world, we're only doing data mapping here. Where it could hurt is if somebody sets it to null. Do you have an annotation to prevent nullable in PHP?

Reply:
Makes sense. Yes, we can easily prevent passing a null for PHP: `setBackingStore(BackingStore $value)`. Nullable would be `?BackingStore`

Reply:
Addressed